nio/rooms.py

```diff
@@ -97,25 +99,31 @@ def group_name(self):
         """
         # Sort user display names, excluding our own user and using the
         # mxid as the sorting key.
-        #
-        # TODO: Hook the user display name disambiguation algorithm here.
-        # Currently, we use the user display names as is, which may not be
-        # unique.
-        users = [
-            user.user_id
-            for mxid, user in sorted(self.users.items(), key=lambda t: t[0])
-            if mxid != self.own_user_id
-        ]
-
-        num_users = len(users)
+
+        names = defaultdict(list)
+        users = {}
+        for mxid, user in self.users.items():
```

Comment:
This is sub-optimal, it would be better to cache the conflicting names so we don't need to do this calculation every time we ask for the room display name.

Another advantage of caching them is that weechat can easily check if there is a conflict instead of calculating it again itself.

Reply:
it's easy to forget that 2k users in a single room isn't a theoretical issue but a reality.